src/components/Modals/BuyItemSlotsModal/BuyItemSlotsModal.tsx

```diff
@@ -76,32 +62,30 @@ export default class BuyItemSlotsModal extends React.PureComponent<Props, State>
           <span>{t('buy_item_slots_modal.description_line_two')}</span>
         </ModalDescription>
         <Modal.Content className={styles.content}>
-          {(isFetchingTiers || tiers === undefined) && (
-            <div className={styles.loader}>
-              <Loader active size="large" />
-            </div>
-          )}
-          <div>
-            {tiers.sort(sortTiers).map(tier => (
-              <div className={styles.tier} key={tier.id}>
-                <div className={styles.tierType}>
-                  <Radio
-                    name="tier-radio"
-                    disabled={isBuyingItemSlots}
-                    value={tier.id}
-                    checked={tier.id === selectedTierId}
-                    onChange={this.handleTierChange}
-                  />
-                  <span>{t('buy_item_slots_modal.tier_value', { value: Number(tier.value).toLocaleString() })}</span>
+          <div className={styles.slotsPriceContainer}>
+            {isFetchingSlotPrice ? (
+              <Loader active size="tiny" />
+            ) : (
+              <>
+                <Field label={t('buy_item_slots_modal.how_many_slots_title')} placeholder="1" onChange={this.handleSlotToBuyChange} />
+                <div className={styles.slotValue}>
+                  {t('buy_item_slots_modal.slots_value', {
+                    symbol: <Mana network={Network.MATIC} size="small" />,
+                    slot_cost: slotPrice,
+                    total_cost: slotPrice && slotsToBuy ? slotPrice * Number(slotsToBuy) : 0
+                  })}
                 </div>
-                <div className={styles.tierValue}>
-                  <Mana network={Network.MATIC} inline /> {Number(fromWei(tier.price, 'ether')).toLocaleString()}
+                <div className={styles.slotValue}>
+                  {t('buy_item_slots_modal.total_cost', {
+                    symbol: <Mana network={Network.MATIC} size="small" />,
+                    total_cost: slotPrice && slotsToBuy ? slotPrice * Number(slotsToBuy) : 0
```

Comment:
As we can put an arbitrary amount here, what if we work with big numbers so we don't get the scientific notation to pop up.

Reply:
Good idea! Updated to use `BigNumber` :) 

<img width="610" alt="image" src="https://user-images.githubusercontent.com/8763687/154126924-d9b42ec3-34ea-452b-9f9f-c326d46be8ce.png">